cli/flox/src/commands/environment.rs

```diff
@@ -1198,6 +1199,13 @@ impl Push {
                 To rename your environment: 'flox edit --name <new name>'
                 To pull and manually re-apply your changes: 'flox delete && flox pull -r {owner}/{name}'
             "}.into(),
+            ManagedEnvironmentError::Build(_) => formatdoc! {"
+                ❌  Unable to push environment with compatibility errors on your system ({system}).
+
+                {err}
```

Comment:
The actual error still isn't getting printed:
```
ERROR: ❌  Unable to push environment with compatibility errors on your system (aarch64-darwin).

    could not build environment

    Use 'flox edit' to resolve errors, test with 'flox activate', and 'flox push' again.
```

Reply:
Well it is just not the helpful bits...
Fixing this tomorrow
Just wondering if we should display the entire chain or not just the root cause (esp. for these pkgdb ones).
Leaning towards the latter

Reply:
I bet we'll come back and change it anyways when we rework errors, so either's probably fine for now